### PR TITLE
DEP: stats: removal of kwargs n in stats.moment and alpha in stats.interval

### PR DESCRIPTION
**Pull request by j-bowhay**
#### Reference issue

follow up to https://github.com/scipy/scipy/pull/13490 / #16389
#### What does this implement/fix?

Removes deprecated kwargs now development for 1.11 has started

### CODE REVIEW

File: scipy/stats/_distn_infrastructure.py [hunk 1]
```diff
@@ -1209,14 +1209,9 @@ def entropy(self, *args, **kwds):
         place(output, cond0, self.vecentropy(*goodargs) + log(goodscale))
         return output[()]
```
Comment (mdhaber) [inline scipy/stats/_distn_infrastructure.py:1211]: Please mak these changes up here, too:
https://github.com/scipy/scipy/blob/d6b06f5ecc7f09d205b2fd2ce17c4254b4582f07/scipy/stats/_distn_infrastructure.py#L493-L500

(see how they were added in 13490 - I think `*args` and `*kwds` can be removed from the signature up there.)

File: scipy/stats/_distn_infrastructure.py [hunk 2]
```diff
@@ -1230,87 +1225,6 @@ def moment(self, order=None, *args, **kwds):
...
-                raise TypeError(message)
-            else:  # B2
-                message = ("Use of keyword argument 'n' for method 'moment is"
-                           " deprecated and will be removed in SciPy 1.11.0. "
-                           "Use first positional argument or keyword argument"
-                           " 'order' instead.")
-                order = kwds.pop("n")
-                warnings.warn(message, DeprecationWarning, stacklevel=2)
         n = order
         # No special treatment of A1, A3, or B3 is needed because the order
```
Comment (mdhaber) [inline scipy/stats/_distn_infrastructure.py:1229]: This comment can be removed. (Comparing against https://github.com/scipy/scipy/pull/13490 is how I'm finding these.)

**Review by mdhaber — COMMENTED** [review]
LGTM aside from these things.

**Comment by mdhaber** [comment]
Thanks @j-bowhay!

**Comment by j-bowhay** [comment]
Thanks for the review @mdhaber, do you mind marking this a completed in #15765?

**Comment by mdhaber** [comment]
Done. Please feel free to @mention me in any other stats removals.

**Comment by h-vetinari** [comment]
I added a [release note](https://github.com/scipy/scipy/wiki/*Release-Note-Entries-for-SciPy-1.11.0-(asterisk-makes-this-appear-at-the-top)#expired-deprecations) for this (and your other PRs), please double-check :)

**Comment by j-bowhay** [comment]
@h-vetinari thank you that is very helpful as now only maintainers can add release notes

**Pull request merged**